src/meta/src/hummock/compaction/mod.rs

```diff
@@ -336,9 +284,37 @@ impl LocalSelectorStatistic {
 
 pub trait CompactionPicker {
     fn pick_compaction(
-        &self,
+        &mut self,
```

Comment:
This change indicates that the picker itself becomes stateful across multiple compaction runs, which seems more complicated to maintain than before. Since currently only `SpaceReclaimCompactionPicker` and `TtlReclaimCompactionPicker` are stateful, I am thinking whether it is neecessary to modify the trait interface.